### DISCUSSION TITLE
Some errors happened in the evaluate mode in test.py

### DISCUSSION THREAD

**Discussion by BugMaker2002** [Dylan-H-Wang/SLF-RPM#11] [Q&A] [answered]
**When I try to run the following command for evaluate mode in test.py:**
python test.py --evaluate --gpu 0 --epochs 100 --batch_size 128 --lr 5e-3 --dropout 0 \
	--pretrained "./logs/ubfc/train/ubfc_best.pth" \
	--dataset_name "ubfc-rppg" \
	--dataset_dir "/share2/data/zhouwenqing/UBFC_rPPG_processed" \
	--workers 4 --vid_frame 150 --vid_frame_stride 2 \
	--log_dir "./logs/ubfc/test" \
	--model_depth 18
**Some errors happened:**
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/58eb229f-a325-4154-9e81-c2622243b88a)
**However, if I remove the evaluate parameter from the command like this: the program runs without error**
python test.py --gpu 0 --epochs 100 --batch_size 128 --lr 5e-3 --dropout 0 \
	--pretrained "./logs/ubfc/train/ubfc_best.pth" \
	--dataset_name "ubfc-rppg" \
	--dataset_dir "/share2/data/zhouwenqing/UBFC_rPPG_processed" \
	--workers 4 --vid_frame 150 --vid_frame_stride 2 \
	--log_dir "./logs/ubfc/test" \
	--model_depth 18

**Comment by Dylan-H-Wang** [comment, marked as answer]
Because the provided weights are SSL-pre-trained weights and it is expected to be fine-tuned on some datasets.
The '--evaluate' flag is used for inference which means the weights for`--pretrained` should be the result of fine-tuned model.
In a word, run `test.py` with provided pre-trained weights without `--evaluate` flag first. Then, you can run `test.py` with '--evaluate' using the output weights from the last `test.py` run.
